include/eve/detail/function/simd/common/sum.hpp

```diff
@@ -18,61 +18,70 @@
 
 namespace eve::detail
 {
-  template<arithmetic_scalar_value T, typename N>
-  EVE_FORCEINLINE auto sum_ ( EVE_SUPPORTS(cpu_)
-                            , splat_type const&, wide<T,N> const &v
+  template<callable_options O, arithmetic_scalar_value T, typename N>
```

Comment:
What happened in this file?

Reply:
Because the splatting mode and mask are now passed as callable options and not as function parameters anymore, I merged everything into two functions (splat & no-splat).
I also changed the code of the non-splat-wide-emulated `sum` impl to use our `for_` construct instead of an IILF + fold.